### PR TITLE
fix: correct typing relayer client

### PR DESCRIPTION
**Pull request by 0xsambugs**
Hey OZ team!

We've recently started working closely with OZ Defender at Mean Finance. In particular, we've tried to automate all our entities (sentinels, autotasks, etc) management through the different clients. NChamo explains it better [here](https://forum.openzeppelin.com/t/what-are-your-needs-on-defender-user-permissions/29244/5?u=nchamo).

Anyways, we realized that the `RelayerClient.list` was returning the wrong type. We casted it to the correct type in our code, but we thought it made sense to fix it for everyone else.

`list` is basically just an API call, but it was returning a class. We think this fixes it, but let us know if we made a mistake or something.

Anyways, keep up the amazing work!

### CODE REVIEW

**Review by ernestognw — APPROVED** [review]
Hey @0xsambugs, thank you very much for the contribution! We love to see the community engaging with our open-source code 🙌

I tested on local and it doesn't change the interface while doing a `console.log`, so I think this is fine and it's not introducing breaking changes.

I'll give my LGTM but @mok0230 is more familiar with this code so I'll wait for his approval before merging.

**Comment by mok0230** [comment]
Looks great, really appreciate the contribution @0xsambugs !

**Pull request merged**
